CMake/itkExternal_FFTW.cmake

```diff
@@ -12,153 +30,152 @@ if(NOT ITK_USE_MKL AND NOT ITK_USE_CUFFTW)
-    message(ERROR "install fftw and use ITK_USE_SYSTEM_FFTW")
-  else()
+if(NOT ITK_USE_SYSTEM_FFTW)
     #
     # fftw limitation -- can't be built in
     # a directory with whitespace in its name.
     if(${CMAKE_CURRENT_BINARY_DIR} MATCHES ".*[ \t].*")
       message(FATAL_ERROR
         "Can't build fftw in a directory with whitespace in its name")
     endif()
-    #
-    # build fftw as an external project
-    if(BUILD_SHARED_LIBS)
-      set(FFTW_SHARED_FLAG --enable-shared)
-    endif()
-    #
-    # set fPIC flag if needed
-    set(GCC_POSITION_INDEPENDENT_CODE_FLAG "")
-    if(CMAKE_POSITION_INDEPENDENT_CODE)
-      set(GCC_POSITION_INDEPENDENT_CODE_FLAG "-fPIC")
+
+    if(WIN32)
+      # on windows the build of FFTW must match that of ITK
+      set(FFTW_BUILD_TYPE ${CMAKE_BUILD_TYPE})
+    else()
+      set(FFTW_BUILD_TYPE Release)
     endif()
 
+    include(GNUInstallDirs)
+
     set(_fftw_target_version 3.3.8)
     set(_fftw_url_hash "ab918b742a7c7dcb56390a0a0014f517a6dff9a2e4b4591060deeb2c652bf3c6868aa74559a422a276b853289b4b701bdcbd3d4d8c08943acf29167a7be81a38")
     set(_fftw_url "https://data.kitware.com/api/v1/file/hashsum/sha512/${_fftw_url_hash}/download")
 
+    set(FFTW_STAGED_INSTALL_PREFIX "${ITK_BINARY_DIR}/fftw")
+    set(PROJ_FFTWD_DEPENDS "")
     if(ITK_USE_FFTWF)
       itk_download_attempt_check(FFTW)
       ExternalProject_add(fftwf
         PREFIX fftwf-${_fftw_target_version}
+        INSTALL_DIR ${FFTW_STAGED_INSTALL_PREFIX}
         URL ${_fftw_url}
         URL_HASH SHA512=${_fftw_url_hash}
         DOWNLOAD_NAME "fftw-${_fftw_target_version}.tar.gz"
-        CONFIGURE_COMMAND
-          env
-            "CC=${CMAKE_C_COMPILER_LAUNCHER} ${CMAKE_C_COMPILER}"
-            "CFLAGS=${CMAKE_C_FLAGS} ${CMAKE_C_FLAGS_RELEASE} ${GCC_POSITION_INDEPENDENT_CODE_FLAG} ${_additional_deployment_target_flags}"
-            "LDFLAGS=$ENV{LDFLAGS}"
-            "LIBS=$ENV{LIBS}"
-            "CPP=$ENV{CPP}"
-            "CPPFLAGS=$ENV{CPPFLAGS} ${_additional_deployment_target_flags}"
-            "CXXFLAGS=$ENV{CXXFLAGS} ${GCC_POSITION_INDEPENDENT_CODE_FLAG} ${_additional_deployment_target_flags}"
-            ${_additional_configure_env}
-          ${ITK_BINARY_DIR}/fftwf-${_fftw_target_version}/src/fftwf/configure
-            ${FFTW_SHARED_FLAG}
-            ${FFTW_OPTIMIZATION_CONFIGURATION}
-            ${FFTW_THREADS_CONFIGURATION}
-            --disable-fortran
-            --enable-float
-            --prefix=${ITK_BINARY_DIR}/fftw
-        ${_additional_external_project_args}
+        DOWNLOAD_NO_PROGRESS 1
+        UPDATE_COMMAND ""
+        LOG_CONFIGURE 1
+        LOG_BUILD 1
+        LOG_INSTALL 1
+        CMAKE_CACHE_ARGS
+           -DBUILD_SHARED_LIBS:BOOL=${BUILD_SHARED_LIBS}
+           -DBUILD_TESTS:BOOL=OFF
+           -DCMAKE_BUILD_TYPE:STRING=${FFTW_BUILD_TYPE}
+           -DCMAKE_INSTALL_PREFIX:PATH=${FFTW_STAGED_INSTALL_PREFIX}
+           -DDISABLE_FORTRAN:BOOL=OFF
```

Comment:
Only FORTRAN has a disable prefix?

Reply:
yes. for the cmake interface.

Reply:
I did not notice that earlier, and the new version has DISABLE_FORTRAN:BOOL=ON.